### PR TITLE
Clean out tags when attempting to speak/whisper/emote

### PR DESCRIPTION
**Pull request by FairlySadPanda**
## About the PR
A few bugs have popped up where (legal) string formatting tags have been duplicated out into whisper and local.

Using the RemoveMarkup static we can sanitize attempted speech to remove markup.

## Why / Balance
In some instances legal formatting tags can leak into areas where they are not supported - i.e chatting, whispering and emoting. See: https://github.com/space-wizards/space-station-14/issues/23018

SS14 currently does not support players being able to format their own speech via tags - I'd suspect this would be a candidate for revision during a refactor of ChatSystem. (Being able to emphasise a word or two would allow for easier RP - "I _really think_ you should not open that airlock") As such there's never a reason why a player, or something emulating a player speaking like an intercom, should be posting a tag.

## Technical details
When handling whispers we sanitize out the tags at the top of the method. This is both because DRY and allows for obfuscation to work without causing a thrown exception during text parsing, a weird bug that I will leave of interest only to historians.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
None, as far as known

**Changelog**
:cl:
- fix: Talking, whispering and emoting will now strip out any markup tags in the input text. This also means intercoms tuned to the science channel will no longer squawk out "[bold]", and so on, when reporting on research unlocks.

### CODE REVIEW

**Comment by FairlySadPanda** [comment]
Hi hi, first PR.

Text sanitization like this almost certainly has edge case issues but I'm happy to do the legwork to test and do further fixes. Also I'm horrendous at changelog entries, so apologies if the one I've written is rubbish.

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 1]
```diff
@@ -614,7 +614,8 @@ private void SendDeadChat(EntityUid source, ICommonSession player, string messag
 
     #region Utility
 
-    private enum MessageRangeCheckResult {
+    private enum MessageRangeCheckResult
+    {
```
Comment (FairlySadPanda) [inline Content.Server/Chat/Systems/ChatSystem.cs:618]: There's some instances of my linter cleaning things up - I've seen guidance on PRs here that fixes should be minimal here, so let me know if I need to clonk my linter on the head.

Comment (AJCM-git) [reply Content.Server/Chat/Systems/ChatSystem.cs:618]: Eh, we do allow some level of formatting (and warnings cleanup for that matter) IF it follows our usual guidelines/style AND there are changes other than formatting AND the formatting changes aren't 90% of the PR, so this is very much allowed

Comment (AJCM-git) [reply Content.Server/Chat/Systems/ChatSystem.cs:618]: Still should prob vibe check your formatter just so the third case I described doesn't happens in the future

**Review by AJCM-git — APPROVED** [review]
Looks fine, will test in a couple of days prob tho
